pkg/asset/installconfig/installconfig.go

```diff
@@ -0,0 +1,107 @@
+package installconfig
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/ghodss/yaml"
+	"github.com/pborman/uuid"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/types"
+)
+
+// InstallConfig generates the install-config.yml file.
+type installConfig struct {
+	assetStock  Stock
+	directory   string
+	inputReader *bufio.Reader
+}
+
+var _ asset.Asset = (*installConfig)(nil)
+
+// Dependencies returns all of the dependencies directly needed by an
+// installConfig asset.
+func (a *installConfig) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		a.assetStock.EmailAddress(),
+		a.assetStock.Password(),
+		a.assetStock.BaseDomain(),
+		a.assetStock.ClusterName(),
+		a.assetStock.License(),
+		a.assetStock.PullSecret(),
+		a.assetStock.Platform(),
+	}
+}
+
+// Generate generates the install-config.yml file.
+func (a *installConfig) Generate(dependencies map[asset.Asset]*asset.State) (*asset.State, error) {
+	emailAddress := string(dependencies[a.assetStock.EmailAddress()].Contents[0].Data)
+	password := string(dependencies[a.assetStock.Password()].Contents[0].Data)
+	baseDomain := string(dependencies[a.assetStock.BaseDomain()].Contents[0].Data)
+	clusterName := string(dependencies[a.assetStock.ClusterName()].Contents[0].Data)
+	license := string(dependencies[a.assetStock.License()].Contents[0].Data)
+	pullSecret := string(dependencies[a.assetStock.PullSecret()].Contents[0].Data)
+
+	installConfig := types.InstallConfig{
+		ObjectMeta: metav1.ObjectMeta{
```

Comment:
Why do we need objectmeta if it's just for ClusterName?
I am not strongly against this though, but might be good if we can add some docs in the object, so it's easier for other folks to find the ClusterName

Reply:
I don't know what the plans are for objectmeta. It was defined that way in https://github.com/openshift/installer/blob/master/Documentation/design/installconfig.md. I also don't have a strong opinion one way or the other whether the Cluster Name is placed in the objectmeta or in a separate field.